sdks/java/io/amazon-web-services/src/test/java/org/apache/beam/sdk/io/aws/s3/S3FileSystemTest.java

```diff
@@ -107,59 +106,64 @@ public static void afterClass() {
 
   @Test
   public void testGetScheme() {
-    S3FileSystem s3FileSystem = new S3FileSystem(s3Options());
+    S3FileSystem s3FileSystem = new S3FileSystem(s3Config("s3"));
```

Comment:
Would it better to keep using `s3Options()` here to make sure that schema is properly parsed from options as well?

In general, I'd prefer to keep all old tests as they were before to make sure that we don't have any user API breakage.

Reply:
OK, I've done this as much as possible by adding alternate versions of the tests called e.g. testGetSchemaWithS3Options. These follow the original test code closely, although S3ResourceIds have to be constructed with an explicit scheme of "s3" in these cases.